src/hipscat_import/soap/run_soap.py

```diff
@@ -22,28 +22,50 @@ def run(args, client):
     resume_plan = SoapPlan(args)
     if not resume_plan.is_counting_done():
         futures = []
-        for source_pixel, object_pixels, _source_key in resume_plan.count_keys:
+        for source_pixel, object_pixels, source_key in resume_plan.count_keys:
             futures.append(
                 client.submit(
                     count_joins,
+                    key=source_key,
                     soap_args=args,
                     source_pixel=source_pixel,
                     object_pixels=object_pixels,
-                    cache_path=args.tmp_path,
                 )
             )
```

Comment:
hey @delucchi-cmu, last week you removed the task’s key here because it was causing issues with the full pipeline execution (#174). Is it supposed to be back, or was it readded by mistake?

Reply:
Great catch! That was re-added by copy pasta mistake.